handlers/event.js

```diff
@@ -88,6 +87,73 @@ module.exports.get = async (event, ctx, callback) => {
 
 };
 
+module.exports.getUsers = async (event, ctx, callback) => {
+
+  const id = event.queryStringParameters.id;
+
+  // Check that parameters are valid
+  if (!id) {
+    callback(null, helpers.inputError('id not specified.', 'missing query param'));
+  }
+
+  const params = {
+    TableName: 'biztechRegistration' + process.env.ENVIRONMENT,
+    FilterExpression: 'eventID = :query',
+    ExpressionAttributeValues: {
+      ':query': id
+    }
+  };
+
+  await docClient.scan(params).promise()
+  .then(async result => {
+    console.log('Scan success.');
+    const registrationList = result.Items;
+
+    /**
+     * Example registration obj:
+     * { eventID: 'blueprint',
+     *   id: 123,
+     *   updatedAt: 1580007893340,
+     *   registrationStatus: 'registered'
+     * }
+     */
+    let keysForRequest = registrationList.map(registrationObj => {
+      let keyEntry = {}
+      keyEntry.id = parseInt(registrationObj.id)
+      return keyEntry
+    })
+    console.log('Keys:', keysForRequest)
+
+    let keyBatches = [];
+    const size = 100 // max BatchGetItem count
+    while (keysForRequest.length > 0) {
+      keyBatches.push(keysForRequest.splice(0, size))
+    }
+
+    await Promise.all(keyBatches.map(batch => {
+      return helpers.batchGet(batch, 'biztechUsers' + process.env.ENVIRONMENT)
+    })).then(result => {
+      const results = result.flatMap(batchResult => batchResult.Responses.biztechUsers)
```

Comment:
So `flatMap` is being used to get the data of each user in the list of batches

Reply:
We have an array of batch calls, which returns an array of promises, when evaluated we have an array of arrays of batch get response data. Flatmap combines that into an array of user data